### ISSUE TITLE
Add support for explainer

### ISSUE THREAD

**Issue by bukson** [catboost/catboost#445]
For xgboost and lighgbm there are several explainer packages (https://github.com/lantanacamara/lightgbmExplainer, https://github.com/AppliedDataSciencePartners/xgboostExplainer) is it possible to prepare some explainer for catboost to be able to check predictions for particular examples?

**Comment by annaveronika** [comment]
We have shap values, please, check the tutorial https://github.com/catboost/tutorials/blob/master/advanced_tutorials/shap_values_tutorial.ipynb

Using this functionality you can see the impact of each feature on the prediction.

We also have object importance, see this tutorial:
https://github.com/catboost/tutorials/blob/master/advanced_tutorials/catboost_object_importance_tutorial.ipynb

It allows to see influence of objects from train dataset on the error on your object.

**Comment by annaveronika** [comment]
Closing the issue, please feel free to reopen, if you will still have questions.

**Comment by bukson** [comment]
Thank you for your answer. I am aware of existance of shap values, but I don't think it is relevant to my issue. I am not intrested in explaining impact of each feature on whole dataset, but on particular example, pleasecompare to https://medium.com/applied-data-science/new-r-package-the-xgboost-explainer-51dd7d1aa211 

>  -1.41 ::: Baseline (Intercept) 
>- 1.10 ::: Satisfaction Level (prediction is now -2.51) 
> +0.98 ::: Last Evaluation (prediction is now -1.53) 
> +0.32 ::: Time Spent At Company (prediction is now - 1.21) 
> +0.27 ::: Hours Average Monthly (prediction is now -0.94) 
>- 0.24 ::: Sales (prediction is now -1.18) 
>- 0.18 ::: Number of Projects (prediction is now -1.36)
> +0.11 ::: Work Accident (prediction is now -1.25) 
> -0.07 ::: Salary (prediction is now -1.32) 
> +0.02 ::: Promotion Last 5 Years (prediction is now -1.30)

> = -1.299 ::: Prediction

> The probability is then just the logistic function applied to the log-odds.

> 1/(1+exp(-(-1.299))) = 0.21433

I checked both of tutorials and I don't know how i can  come from shap values straight to probability.

**Comment by bukson** [comment]
Byw i can't reopen if you close issue :)

**Comment by bukson** [comment]
Ok I can see it but how you came from Formula raw prediction to Probability of class 1 in [11]?

**Comment by annaveronika** [comment]
Probability is a sigmoid of the raw value.

It is explained well in https://github.com/catboost/tutorials/blob/master/python_tutorial_with_tasks.ipynb
In [36] - In [39] show what are raw values.
In [49] - In [50] show how to do explanation on an object

**Comment by bukson** [comment]
Ok thank you very much!
